### PR TITLE
test(rbno): Add longevities for repair based node operations

### PR DESCRIPTION
**Pull request by roydahan**
Adding several longevities to test RBNO feature.
Four types of nemesis are being used to test it:
1. Remove Node and bootstrap new node instead.
2. Replace a dead node.
3. Decommission and add new node.
4. Grow cluster by adding nodes then shrink it back with decommission

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by roydahan** [comment]
@juliayakovlev this is the PR with all the RBNO jobs.
For now you need to use only those for "ReplaceNode".
I suggest you will run 1-2 of them in staging, making sure they work and then we merge them and run them in 4.6.

**Comment by roydahan** [comment]
I need to change the longevities to use only the "incremental" changes in the configuration.

**Comment by roydahan** [comment]
@juliayakovlev I changed all yamls to be "incremental" (only changing what needed on top of the original longevity).
Also, I changed the pipelines accordingly.

Please test RBNO feature for replace node with it.
Run it once from this branch, if it's ok, we will merge and you can run the official jobs from official branch.

File: test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+nemesis_class_name: 'NodeTerminateAndReplace'
+nemesis_interval: 2
+nemesis_filter_seeds: false
```
Comment (juliayakovlev) [inline test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:3]: @roydahan Why did you set `nemesis_filter_seeds` to false? "longevity-50GB-3days-authorization-and-tls-ssl.yaml" set 1 seed node only

Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:3]: I don't remember, but if it's a problem we can set to true or better increase the number of seeds.

Comment (juliayakovlev) [reply test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:3]: Yes, we can't run terminate and replace on seed if one seed in the cluster only.
I created my branch, set seed_num to 3 and run the test with my branch. Can you, please, fix it here

Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:3]: you can push to this PR

**Comment by juliayakovlev** [comment]
@roydahan Roy, what is test duration should be? You called the tests "4h", but actually the test duration set in`longevity-50GB-3days-authorization-and-tls-ssl`, that 3 days

**Comment by roydahan** [comment]
> @roydahan Roy, what is test duration should be? You called the tests "4h", but actually the test duration set in`longevity-50GB-3days-authorization-and-tls-ssl`, that 3 days

where?

File: test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+nemesis_class_name: 'TerminateAndRemoveNodeMonkey'
```
Comment (bentsi) [inline test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml:1]: missing `test_duration: 240`

**Review by bentsi — CHANGES_REQUESTED** [review]

File: jenkins-pipelines/repair-based-operation/longevity-cdc-4h-ReplaceNode.jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+longevityPipeline(
+    backend: 'aws',
+    aws_region: 'eu-west-1',
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml", "test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-ReplaceNode.yaml"]''',
+
+    timeout: [time: 360, unit: 'MINUTES']
```
Comment (bentsi) [inline jenkins-pipelines/repair-based-operation/longevity-cdc-4h-ReplaceNode.jenkinsfile:12]: not needed

File: jenkins-pipelines/repair-based-operation/longevity-100gb-4h-ReplaceNode.jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+longevityPipeline(
+    backend: 'aws',
+    aws_region: 'eu-west-1',
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml", "test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml"]''',
+
+    timeout: [time: 360, unit: 'MINUTES']
```
[the remark this answers is not in the document]
Comment (roydahan) [reply jenkins-pipelines/repair-based-operation/longevity-100gb-4h-ReplaceNode.jenkinsfile:12]: This is how almost all our pipeline looks like.
If we want we need to send a PR and remove from all of them.

File: test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+nemesis_class_name: 'NodeTerminateAndReplace'
```
[the remark this answers is not in the document]
Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:1]: The test_duration is in the "base" yaml (each pipeline has two yamls).
This yaml is only overriding the base yamls with some parameters.
There was a mistake in the base yaml that used incorrect yaml of a longer run, now it's fixed.

**Comment by roydahan** [comment]
@bentsi & @juliayakovlev, I added "seed_num: 3" to the "ReplaceNode" longevities that has "nemesis_filter_seed: false"
It should do.
I also fixed a mistake where some of the pipelines used the wrong base yaml.

File: jenkins-pipelines/repair-based-operation/longevity-100gb-4h-ReplaceNode.jenkinsfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (fgelcer) [reply jenkins-pipelines/repair-based-operation/longevity-100gb-4h-ReplaceNode.jenkinsfile:12]: yes, it need to be removed from all `jenkinsfile` as it is not used anymore

File: test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+nemesis_interval: 5
+nemesis_during_prepare: false
```
Comment (juliayakovlev) [inline test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml:3]: Please, add `seeds_num` or chande `nemesis_filter_seeds` to True
"longevity-cdc-100gb-4h.yaml" runs with 1 seed.

**Review by juliayakovlev — CHANGES_REQUESTED** [review]

File: test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-Decommission.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+nemesis_class_name: 'DecommissionMonkey'
+nemesis_interval: 5
+nemesis_during_prepare: false
```
[the remark this answers is not in the document]
Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-Decommission.yaml:3]: 1. Why is it needed here?
2. We don't care now about Decommission operation.

File: test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml (hunk 6 above)
Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml:3]: Same as above.

File: test-cases/features/repair-based-operations/longevity-50GB-12h-ReplaceNode.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+test_duration: 760
+
+stress_cmd: ["cassandra-stress mixed cl=QUORUM duration=11h -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native  -rate threads=100 -pop seq=1..100000000 -log interval=5",
+             "cassandra-stress user profile=/tmp/cs_mv_profile.yaml ops'(insert=3,read1=1,read2=1,read3=1)' cl=QUORUM duration=11h -port jmx=6868 -mode cql3 native -rate threads=20"]
+
+stress_read_cmd: ["cassandra-stress read cl=QUORUM duration=11h -port jmx=6868 -mode cql3 native  -rate threads=50 -pop seq=1..100000000 -log interval=5"]
+
+nemesis_class_name: 'NodeTerminateAndReplace'
+nemesis_interval: 5
+nemesis_during_prepare: false
```
[the remark this answers is not in the document]
Comment (roydahan) [reply test-cases/features/repair-based-operations/longevity-50GB-12h-ReplaceNode.yaml:10]: we don't have  nemesis_filter_seeds here.

[the remark this answers is not in the document]
Comment (juliayakovlev) [reply test-cases/features/repair-based-operations/longevity-50GB-12h-ReplaceNode.yaml:10]: my mistake

File: test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-Decommission.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (juliayakovlev) [reply test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-Decommission.yaml:3]: my mistake

**Comment by bentsi** [comment]
@roydahan please fix pre-commit
image

**Comment by roydahan** [comment]
@bentsi passed all checks

**Pull request merged**